EIPS/eip-draft-whisper.md

```diff
@@ -81,35 +79,64 @@ PoW calculation:
+
+The projection function is defined as a mapping from a a 4-byte slice S to a 512-bit slice D; for ease of explanation, S will dereference to bytes, whereas D will dereference to bits.
+
+	LET D[*] = 0
+	FOREACH i IN { 0, 1, 2 } DO
+	LET n = S[i]
+	IF S[3] & (2 ** i) THEN n += 512
+	D[n] = 1
+	END FOR
+
+
+OPTIONAL
+
+**P2P Request** [`126`, `whisper_envelope`]
+
+This packet contains two objects: integer (0x7E) followed by a single Whisper Envelope.
+
+This packet is used for sending Dapp-level peer-to-peer requests, e.g. Whisper Mail Client requesting old messages from the Whisper Mail Server.
+
+**P2P Message** [`127`, `whisper_envelope`]
+
+This packet contains two objects: integer (0x7F) followed by a single Whisper Envelope.
+
+This packet is used for sending the peer-to-peer messages, which are not supposed to be forwarded any further. E.g. it might be used by the Whisper Mail Server for delivery of old (expired) messages, which is otherwise not allowed.
+
+
 ### Whisper Envelope
 
 Envelopes are RLP-encoded structures of the following format:
 
-	[ Version, Expiry, TTL, Topic, AESNonce, Data, EnvNonce ]
+	[ Version, Expiry, TTL, Topic, Data, Nonce ]
 	
-`Version`: up to 4 bytes (currently one byte containing zero). Version indicates encryption method. If Version is higher than current, envelope could not be decoded, and therefore only forwarded to the peers.
+`Version`: 1 byte. If Version is higher than current, envelope could not be decoded, and therefore only forwarded to the peers.
 
 `Expiry`: 4 bytes (UNIX time in seconds).
 
 `TTL`: 4 bytes (time-to-live in seconds).
 
 `Topic`: 4 bytes of arbitrary data.
 
-`AESNonce`: 12 bytes of random data (only present in case of symmetric encryption).
-
 `Data`: byte array of arbitrary size (contains encrypted message).
 
-`EnvNonce`: 8 bytes of arbitrary data (used for PoW calculation).
+`Nonce`: 8 bytes of arbitrary data (used for PoW calculation).
 
 ### Contents of Data Field (Message)
 
-Data field contains encrypted message of the Envelope. Plaintext (unencrypted) payload is formed as a concatenation of a single byte for flags, additional metadata (as stipulated by the flags) and the actual payload. The message has the following structure:
+This section describes optional description of Data Field to set up an example. It is only relevant if you want to decrypt the incoming message, but any other format would be perfectly valid and must be forwarded to the peers.
+
+Data field contains encrypted message of the Envelope. In case of symmetric encryption, it also contains appended Salt (12 bytes). Plaintext (unencrypted) payload is formed as a concatenation of a single byte for flags, additional metadata (as stipulated by the flags) and the actual payload. The message has the following structure:
```

Comment:
agreed on data field format, but I think it actually belongs in the upcoming whisper RPCs EIP under a "Common Whisper Payload Format" section